data/hai/hai missions.txt

```diff
@@ -1658,3 +1658,145 @@ mission "Strider notification"
+				goto continue
+			label "question 1"
+			`	"Some guy in our dorm told us that there's ghosts haunting <planet>," says another one of the Hai, "and we wanted to call his bluff."`
+			label continue
+			`	"We're planning to look around for an afternoon, see what we can find," one of the human students says, grinning. "Once we're done we'll let you know, and then we can leave."`
+			`	One of the shorter Hai in the back pipes up. "I brought cameras for the trip!" He is ignored by the rest of the group.`
+			choice
+				`	"Is this legal?"`
+				`	"Not my business where you spend your vacations. Come aboard."`
+					goto accept
+			`	The students look at each other, considering this question for what is apparently the first time. "We've heard of other people going there with no problems, as long as they were... discreet about it," says one of the students, trying to assure both you and their fellow classmates. Another mentions having heard something about needing a permit to enter, but he is quickly shushed by the rest of the group.`
+			choice
+				`	"I'm not about to risk getting in trouble for this. Find someone else."`
+					decline
+				`	"Not my business where you spend your vacations. Come aboard."`
+			label accept
+			`	The students race onboard your ship. As they settle into their bunks, you can hear them speculating about what they will find on <planet>.`
+				accept
+	on visit
+		dialog `You look for the group of students you were transporting, but realize that they took a ride on one of your escorts! Better depart and wait for your escorts to arrive in this star system.`
+	on complete
+		payment 8000
+		dialog `Your passengers' journey mainly consisted of trying to scare each other by telling ghost stories and jumping out of dimly-lit corners of your ship. Despite these annoyances, you land at the station with no issues. Your passengers practically jump off your ship after hastily paying you <payment>. After passing the time in your ship for a few hours, you are shocked by a loudspeaker sternly pronouncing something in Hai. You figure you should go back to <planet>'s spaceport to see what's going on. Then again, disassociating yourself from them may save you some trouble with law enforcement.`
+
+
+
+mission "Station Exploration 2"
+	priority
+	name "Bring students home to <planet>"
+	description "Bring a group of Hai and human classmates back home to <destination>."
+	passengers 5
+	source "Darkrest"
+	destination "Greenwater"
+	to offer
+		has "Station Exploration 1: done"
+		not "Station Exploration Abandon Ship: declined"
+	on offer
+		conversation
+			`The spaceport is bathed in red light, and the booming voice continues to play. Your red-faced passengers rush back onto your ship shouting at each other in a mixture of languages, but once they're through the airlock you take off before something worse happens.`
+				launch
```

Comment:
Darkrest has no spaceport. It seems like what you want is to take the on complete dialog from the first mission, make it a part of this on offer conversation, then give a choice to let the kids back in that either launches if you let them in or flees if you don't.

This also has no indication that you should return to Greenwater. Given that the abandon mission can be offered anywhere, should the conversation where you bring the kids with you also be offered anywhere? In order to do this, you could give this mission
```
to complete
	never
```
and have it be failed when you land anywhere in Hai space after this mission, which also triggers whichever mission corresponds to your decision.

Reply:
>Given that the abandon mission can be offered anywhere, should the conversation where you bring the kids with you also be offered anywhere?

You haven't really fulfilled your end of the bargain if you don't bring them back to Greenwater as you promised. I'd rather have it only trigger if you land there with the passengers.
I also added your suggestions in the first half of the comment, but I have no idea if I'm doing it correctly. I'm pretty terrible with mission syntax so I'd appreciate if you could give it a lookover and make sure nothing's broken.